src/Article/Article.ts

```diff
@@ -0,0 +1,41 @@
+// todo move logic here
+
+import { articleDetailXId } from '../stores';
+import { ArticleMcsRenderer } from './ArticleMcsRenderer';
+import { ArticleDesktopRenderer } from './ArticleDesktopRenderer';
+
+
+export interface ArticleRenderingOptions {
+  nodet: boolean;
+  strings: KVS<string>;
+  isMainPage: boolean;
+}
+
+
+export class Article {
+  public readonly id: string;
+  public readonly json: any;
+  public details: ArticleDetail;
+  public readonly renderingOptions: ArticleRenderingOptions;
+  private readonly isRendered: boolean;
+
+
+  constructor(id: string, json: any, renderingOptions: ArticleRenderingOptions) {
+    this.id = id;
+    this.json = json;
+    this.renderingOptions = renderingOptions;
+
+    this.isRendered = (
```

Comment:
A value of this flag depends on the `read-only` props and marked as `read-only` itself. In Typescript classes, `read-only` means that the value set in constructor only (once) and never changed. A kind of class-level const. So there's no reasons to convert it to a method because a value will never change during the instance lifecycle.

Reply:
ok